### PR TITLE
Update docs landing page

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates the docs landing page to better describe what `transformers` is, what it offers, and briefly introduce users to its design. I think this gives a clearer picture of `transformers` and is more impactful than listing all the different tasks supported. Let me know what you think!

There's also a minor issue with the image for custom support. Nils is no longer with us, so we may want to update this image with another member of the team. No big deal though :)

### CODE REVIEW

File: docs/source/en/index.mdx [hunk 1]
```diff
@@ -12,18 +12,13 @@ specific language governing permissions and limitations under the License.
 
 # 🤗 Transformers
 
-State-of-the-art Machine Learning for PyTorch, TensorFlow and JAX.
+State-of-the-art Transformers models in PyTorch, TensorFlow, and JAX for research and production.
```
Comment (sgugger) [inline docs/source/en/index.mdx:15]: I wouldn't change the subtitle, which is the same as the README.

File: docs/source/en/index.mdx [hunk 2]
```diff
@@ -12,18 +12,13 @@ specific language governing permissions and limitations under the License.
...
 
-🤗 Transformers provides APIs to easily download and train state-of-the-art pretrained models. Using pretrained models can reduce your compute costs, carbon footprint, and save you time from training a model from scratch. The models can be used across different modalities such as:
+🤗 Transformers is dedicated to providing inference, finetuning, and pretraining for Transformer-based models on text, computer vision, speech, and multimodal tasks. Each model is defined by three classes - a configuration, a model, and a preprocessor - which can be initialized from pretrained instances stored on the [Hugging Face Hub](https://huggingface.co/models). The library also offers convenient helper tools like a simple pipeline to use a model for inference on a given task and a feature-complete training loop optimized for 🤗 Transformers models.
```
Comment (sgugger) [inline docs/source/en/index.mdx:17]: This is not an improvement compared to before IMO. In particular, I liked the list.

File: docs/source/en/index.mdx [hunk 3]
```diff
@@ -12,18 +12,13 @@ specific language governing permissions and limitations under the License.
...
 
-* 📝 Text: text classification, information extraction, question answering, summarization, translation, and text generation in over 100 languages.
-* 🖼️ Images: image classification, object detection, and segmentation.
-* 🗣️ Audio: speech recognition and audio classification.
-* 🐙 Multimodal: table question answering, optical character recognition, information extraction from scanned documents, video classification, and visual question answering.
+Models are available in PyTorch, TensorFlow, and JAX. This framework interoperability grants the flexibility to use a different framework at each stage of a model's lifetime. 🤗 Transformers also supports exporting a model to a format like ONNX and TorchScript for deployment in production environments.
```
Comment (sgugger) [inline docs/source/en/index.mdx:19]: Adding the support to ONNX and TorchScript is good, but I preferred the deleted sentence which explained more clearly the interoperability.

**Review by sgugger — COMMENTED** [review]
Apart from adding the mention we support TorchScript and ONNX, I'm not really a fan of the other changes.

File: docs/source/en/index.mdx (hunk 1 above)
Comment (NielsRogge) [reply docs/source/en/index.mdx:15]: Same comment here for me.

File: docs/source/en/index.mdx (hunk 2 above)
Comment (NielsRogge) [reply docs/source/en/index.mdx:17]: I also prefer the former, to be honest

File: docs/source/en/index.mdx (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/index.mdx:15]: What if I also changed the subtitle in the README? 😏

I'm ok if we don't change the subtitle though, I just feel like this is a slightly more precise description of what `transformers` is. It describes exactly what the library offers and for what purpose

File: docs/source/en/index.mdx (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/index.mdx:17]: What do you think about bringing back most of the list and combining it with a simplified version of the changes introduced here? I think the list is useful for showcasing what the library is capable of, but I think it is also important to briefly describe the library's goals and what it is focused on. I can take out the part about the `pipeline` and `Trainer` here.

File: docs/source/en/index.mdx (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/index.mdx]: Updated to explain the interoperability a little clearer, let me know what you think :)

File: docs/source/en/index.mdx (hunk 1 above)
Comment (sgugger) [reply docs/source/en/index.mdx:15]: The library has multiple models that are not Transformers based, so this new suggestion is factually wrong, not more precise ;-)

File: docs/source/en/index.mdx (hunk 2 above)
Comment (sgugger) [reply docs/source/en/index.mdx:17]: Again, the changes focus too much on Transformer-based models. The goal of the library is perfectly stated in the sentence you are changing (previous version).

File: docs/source/en/index.mdx [hunk 4]
```diff
@@ -12,18 +12,18 @@ specific language governing permissions and limitations under the License.
...
+🤗 Transformers provides APIs and tools to easily download and train state-of-the-art pretrained models. Using pretrained models can reduce your compute costs, carbon footprint, and save you the time and resources required to train a model from scratch. These models support common tasks in different modalities, such as:
 
-* 📝 Text: text classification, information extraction, question answering, summarization, translation, and text generation in over 100 languages.
-* 🖼️ Images: image classification, object detection, and segmentation.
-* 🗣️ Audio: speech recognition and audio classification.
-* 🐙 Multimodal: table question answering, optical character recognition, information extraction from scanned documents, video classification, and visual question answering.
+📝 **Natural Language Processing**: text classification, named entity recognition, question answering, language modeling, summarization, translation, and multiple choice.
```
Comment (sgugger) [inline docs/source/en/index.mdx:19]: Why remove text generation? That's one big application.

File: docs/source/en/index.mdx [hunk 5]
```diff
@@ -12,18 +12,18 @@ specific language governing permissions and limitations under the License.
...
+🖼️ **Computer Vision**: image classification, object detection, and segmentation.
+🗣️ **Audio**: automatic speech recognition and audio classification.
+🐙 **Multimodal**: table question answering, optical character recognition, information extraction from scanned documents, video classification, and visual question answering.
 
-Our library supports seamless integration between three of the most popular deep learning libraries: [PyTorch](https://pytorch.org/), [TensorFlow](https://www.tensorflow.org/) and [JAX](https://jax.readthedocs.io/en/latest/). Train your model in three lines of code in one framework, and load it for inference with another.
+🤗 Transformers support framework interoperability between PyTorch, TensorFlow, and JAX. This provides the flexibility to use a different framework at each stage of a model's life; train a model in three lines of code in one framework, and load it for inference in another. Models can also be exported to a format like ONNX and TorchScript for deployment in production environments.
```
Comment (sgugger) [inline docs/source/en/index.mdx:24]: The links to each framework should be put back, as well as "most popular deep learning libraries:". Some users arrive here as complete beginners and do not necessarily know what those mean.

File: docs/source/en/index.mdx [hunk 6]
```diff
@@ -12,18 +12,18 @@ specific language governing permissions and limitations under the License.
...
 
-Each 🤗 Transformers architecture is defined in a standalone Python module so they can be easily customized for research and experiments.
+Join the growing community on the [Hub](https://huggingface.co/models), [forum](https://discuss.huggingface.co/), or [Discord](https://discord.com/invite/JfAtkvEtRb) today!
```
Comment (sgugger) [inline docs/source/en/index.mdx:26]: Nice addition!

**Review by sgugger — COMMENTED** [review]
Thanks for iterating! Left a couple more comments.

File: docs/source/en/index.mdx (hunk 4 above)
Comment (stevhliu) [reply docs/source/en/index.mdx:19]: Oops that was an accident!

**Review by sgugger — APPROVED** [review]

**Review by patrickvonplaten — APPROVED** [review]
Cool looks nice

**Pull request merged**